tests/kernel/mem_protect/mem_protect/src/mem_domain.c

```diff
@@ -379,3 +379,28 @@ void test_mem_part_overlap(void)
 
 	k_mem_domain_add_partition(&test_domain, &overlap_part);
 }
+
+K_MEM_PARTITION_DEFINE(part_max, ro_buf, sizeof(ro_buf),
+		       K_MEM_PARTITION_P_RW_U_RW);
+
+/**
+ * @brief Test system assert when adding memory partitions more than possible
+ *
+ * @details
+ * - Add memory partitions one by one and more than architecture allows to add.
+ * - When partitions added more than it is allowed by architecture, test that
+ *   system assert for that case works correctly.
+ *
+ * @ingroup kernel_memprotect_tests
+ */
+void test_mem_part_assert_add_overmax(void)
+{
+	int max_parts;
+
+	set_fault_valid(true);
+
+	max_parts = arch_mem_domain_max_partitions_get();
+	for (int i = 0; i <= max_parts; i++) {
+		k_mem_domain_add_partition(&test_domain, &part_max);
```

Comment:
this isn't testing what you think it tests.
memory domain partitions are documented as not being allowed to overlap, so it should be asserting as soon as you try to add the same partition a second time.

Reply:
In another merged PR I already created a test to check system assert when partitions overlap #30004 
Are you sure that we are not talking about that PR #30004 ?

In the current PR the goal is to add partitions more than maximum allowed and to capture a correct error.

Reply:
@andrewboie Could you take a look at?

Reply:
sorry, @ioannisg, @dcpleung, I had some misunderstanding and redesign this case,  the comment from Andrewboie has been addressed in this update. Thanks so much for correcting this.